// wxpay-nodejs/dist/wxpayapi.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _request = require('request');

var _request2 = _interopRequireDefault(_request);

var _security = require('./security');

var _es6Promise = require('es6-promise');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _config = {
    APPID: "",
    MCHID: "",
    KEY: "",
    APPSECRET: "",
    CERT_FILE: "",
    CERT_KEY_FILE: "",
    CA_FILE: "",
    TIMEOUT: 30000
}; /**
    * Created by linyi on 16/8/14.
    */


var WXPayApi = {
    initialize: function initialize(config) {
        _config = config;
    }
};

WXPayApi.makeRequestBody = function (data) {
    var temp = Object.assign({}, data);
    temp.appid = _config.APPID;
    temp.mch_id = _config.MCHID;
    temp.nonce_str = (0, _security.genNonceStr)();
    var body = (0, _security.getSignXML)(temp, _config.KEY);
    console.log("request body:", body);
    return body;
};

WXPayApi.makeRequestPromise = function (options) {
    var promise = new _es6Promise.Promise(function (resolve, reject) {
        _request2.default.post(options, function (error, response, body) {
            if (error) {
                reject(error);
            } else {
                console.log("response body:", body);
                (0, _security.getSafeResult)(body, _config.KEY).then(function (r) {
                    resolve(r);
                }, function (e) {
                    reject(e);
                });
            }
        });
    });
    return promise;
};

/**
 * 统一下单
 * @param data
 * @param timeout
 */
WXPayApi.unifiedorder = function (data, timeout) {
    var options = {
        url: "https://api.mch.weixin.qq.com/pay/unifiedorder",
        body: this.makeRequestBody(data),
        timeout: timeout ? timeout : _config.TIMEOUT
    };
    return this.makeRequestPromise(options);
};

/**
 * 查询订单
 * @param data
 * @param timeout
 */
WXPayApi.orderquery = function (data, timeout) {
    var options = {
        url: "https://api.mch.weixin.qq.com/pay/orderquery",
        body: this.makeRequestBody(data),
        timeout: timeout ? timeout : _config.TIMEOUT
    };
    return this.makeRequestPromise(options);
};

/**
 * 关闭订单
 * @param data
 * @param timeout
 */
WXPayApi.closeorder = function (data, timeout) {
    var options = {
        url: "https://api.mch.weixin.qq.com/pay/closeorder",
        body: this.makeRequestBody(data),
        timeout: timeout ? timeout : _config.TIMEOUT
    };
    return this.makeRequestPromise(options);
};

/**
 * 申请退款,需要证书
 * @param data
 * @param timeout
 */
WXPayApi.refund = function (data, timeout) {
    var options = {
        url: "https://api.mch.weixin.qq.com/secapi/pay/refund",
        body: this.makeRequestBody(data),
        timeout: timeout ? timeout : _config.TIMEOUT,
        cert: fs.readFileSync(_config.CERT_FILE),
        key: fs.readFileSync(_config.CERT_KEY_FILE),
        ca: fs.readFileSync(_config.CA_FILE),
        passphrase: _config.MCHID
    };
    return this.makeRequestPromise(options);
};

/**
 * 查询退款
 * @param data
 * @param timeout
 */
WXPayApi.refundquery = function (data, timeout) {
    var options = {
        url: "https://api.mch.weixin.qq.com/pay/refundquery",
        body: this.makeRequestBody(data),
        timeout: timeout ? timeout : _config.TIMEOUT
    };
    return this.makeRequestPromise(options);
};

exports.default = WXPayApi;
